src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -0,0 +1,972 @@
+                }
+                
+                for (auto & optionalWitness : vInputWitnesses) {
+                    if (!optionalWitness) {
+                        throw JSONRPCError(RPC_WALLET_ERROR, "Witness for note commitment is null");
+                    }
+                    ZCIncrementalWitness w = *optionalWitness; // could use .get();
+                    if (jsChange > 0) {
+                        for (const uint256& commitment : previousCommitments) {
+                            w.append(commitment);
+                        }
+                        if (jsAnchor != w.root()) {
+                            throw JSONRPCError(RPC_WALLET_ERROR, "Witness for spendable note does not have same anchor as change input");
+                        }
+                    }
+                    witnesses.push_back(w);
+                }
+
+                // The jsAnchor is null if this JoinSplit is at the start of a new chain
+                if (jsAnchor.IsNull()) {                   
+                    jsAnchor = inputAnchor;                   
+                }
+
+                // Add spendable notes as inputs
+                std::copy(vInputNotes.begin(), vInputNotes.end(), std::back_inserter(info.notes));
+            }
+
+            
+            //
+            // Find recipient to transfer funds to
+            //            
+            SendManyRecipient smr = zOutputsDeque.front();
+            std::string address = std::get<0>(smr);
+            CAmount value = std::get<1>(smr);
+            std::string hexMemo = std::get<2>(smr);
+            zOutputsDeque.pop_front();
+
+            // Will we have any change?  Has the miners fee been processed yet?
+            jsChange = 0;
+            CAmount outAmount = value;
+            if (!minersFeeProcessed) {
+                if (jsInputValue < minersFee) {
+                    throw JSONRPCError(RPC_WALLET_ERROR, "Not enough funds to pay miners fee");
+                }
+                outAmount += minersFee;
+            }
+            
+            if (jsInputValue > outAmount) {
+                jsChange = jsInputValue - outAmount;
+            } else if (outAmount > jsInputValue) {
+                // Any amount due is owed to the recipient.  Let the miners fee get paid first.
+                CAmount due = outAmount - jsInputValue;
+                SendManyRecipient r = SendManyRecipient(address, due, hexMemo);
+                zOutputsDeque.push_front(r);
+
+                // reduce the amount being sent right now to the value of all inputs
+                value = jsInputValue;
+                if (!minersFeeProcessed) {
+                    value -= minersFee;
+                }
+            }
+            
+            if (!minersFeeProcessed) {
+                minersFeeProcessed = true;
+                info.vpub_new += minersFee; // funds flowing back to public pool
+            }
```

Comment:
It seems as though there could be less conditional code by setting ``minersFee`` to zero once it has been processed, and eliminating the ``minersFeeProcessed`` flag. (Does not block.)